modules/detection/VADTalkMutedDetection.js

```diff
@@ -0,0 +1,298 @@
+         * period voice probability scores.
+         */
+        this._vadEmitter = null;
+
+        /**
+         * Flag which denotes the current state of the detection service i.e.if there is already a processing operation
+         * ongoing.
+         */
+        this._processing = false;
+
+        /**
+         * Buffer that keeps the VAD scores for a period of time.
+         */
+        this._scoreArray = [];
+
+        /**
+         * Promise used to chain create and destroy operations associated with TRACK_ADDED and TRACK_REMOVED events
+         * coming from the conference.
+         * Because we have an async created component (VAD Processor) we need to make sure that it's initialized before
+         * we destroy it ( when changing the device for instance), or when we use it from an external point of entry
+         * i.e. (TRACK_MUTE_CHANGED event callback).
+         */
+        this._vadInitTracker = Promise.resolve();
+
+        /**
+         * {@link JitsiConference} bindings.
+         */
+        conference.on(JitsiConferenceEvents.TRACK_MUTE_CHANGED, this._trackMuteChanged.bind(this));
+        conference.on(JitsiConferenceEvents.TRACK_ADDED, this._trackAdded.bind(this));
+        conference.on(JitsiConferenceEvents.TRACK_REMOVED, this._trackRemoved.bind(this));
+
+        // TODO do we need to handle the case where tracks are removed, make sure this cleans up properly so
+        // we don't have any leeks i.e. stale JitsiLocalTracks
+    }
+
+    /**
+     * Determine if the current score is high enough that we should start the final score processing, and make sure
+     * there isn't already a process operation ongoing.
+     *
+     * @param {number} score - PCM sample VAD score.
+     * @return {boolean}
+     */
+    _shouldStartVADCompute(vadScore) {
+        return vadScore > VAD_VOICE_LEVEL && !this._processing;
+    }
+
+    /**
+     * Determine if the computed score over the configured timestamp should trigger an event.
+     *
+     * @param {number} computedScore - Computed VAD score.
+     * @returns {boolean} - Should or shouldn't trigger.
+     */
+    _shouldTriggerNotification(computedScore) {
+        return computedScore > VAD_AVG_THRESHOLD;
+    }
+
+    /**
+     * Start the {@link TrackVADEmitter} and attach the event listener.
+     * @returns {void}
+     */
+    _startVADEmitter() {
+        this._vadEmitter.on(VAD_SCORE_PUBLISHED, this._processVADScore.bind(this));
+        this._vadEmitter.start();
+    }
+
+    /**
+     * Stop the {@link TrackVADEmitter} and detach the event listener.
+     * @returns {void}
+     */
+    _stopVADEmitter() {
+        this._vadEmitter.removeAllListeners(VAD_SCORE_PUBLISHED);
+        this._vadEmitter.stop();
+    }
+
+    /**
+     * Calculates the average value of a Float32Array.
+     *
+     * @param {Float32Array} scoreArray - Array of vad scores.
+     * @returns {number} - Score average.
+     */
+    _calculateAverage(scoreArray) {
```

Comment:
It could be I just prefer things a bit more verbose so it's easier to read. IMO long single line computations like this one can become harder to read.
But if you feel strongly about it, I can change it.

Reply:
Ok, I'll change it.
Although I don't understand what good does it do to move the function outside the scope of the class if it doesn't use `this` , is it some sort of js best practice? . Functions can also be used to logically group together code blocks for easier readability within a class that's one of the reasons why you have private functions.
I do understand however that this particular function could be placed in some sort of Utility grouping of functions as it's not directly related to the class, but I didn't find anything like that.